`override-data` panic: assignment to entry in nil map

## Describe the bug

Trying to render template with overridden data results in panic.

Initially I have tried with overriding os to `darwin` while running chezmoi on Linux, but it even fails with the non-data info from https://github.com/twpayne/chezmoi/blob/9df5bfa79c4fea7bdbd9f4cb4076da9474d0db7d/internal/cmd/testdata/scripts/issue4634.txtar#L4

## To reproduce

`chezmoi execute-template --override-data '{"foo":"baz"}' '{{ .foo }}'`

## Expected behavior

`baz`

## Output of command with the `--verbose` flag

```console
❯ chezmoi execute-template --override-data '{"foo":"baz"}' '{{ .foo }}' --verbose
panic: assignment to entry in nil map

goroutine 1 [running]:
chezmoi.io/chezmoi/internal/chezmoi.RecursiveMerge(0x0, 0xd?)
	chezmoi.io/chezmoi/internal/chezmoi/recursivemerge.go:25 +0x212
chezmoi.io/chezmoi/internal/cmd.(*Config).newSourceState(0xc0003d5008, {0x1a63ce0, 0x266ab40}, 0xc000494008, {0xc0004c7848, 0x2, 0x5b3bf9?})
	chezmoi.io/chezmoi/internal/cmd/config.go:2031 +0x2cc
chezmoi.io/chezmoi/internal/cmd.(*Config).runExecuteTemplateCmd(0xc0003d5008, 0xc000494008, {0xc00061a500, 0x1, 0x0?})
	chezmoi.io/chezmoi/internal/cmd/executetemplatecmd.go:86 +0x3a6
github.com/spf13/cobra.(*Command).execute(0xc000494008, {0xc00061a4c0, 0x4, 0x4})
	github.com/spf13/cobra@v1.10.1/command.go:1015 +0xb02
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001dc608)
	github.com/spf13/cobra@v1.10.1/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.10.1/command.go:1071
chezmoi.io/chezmoi/internal/cmd.(*Config).execute(0xc0003d5008, {0xc000040250, 0x5, 0x5})
	chezmoi.io/chezmoi/internal/cmd/config.go:1352 +0xbd
chezmoi.io/chezmoi/internal/cmd.runMain({{0x1a48764, 0x6}, {0x1a60100, 0x28}, {0x1a4f960, 0x14}, {0x1a49940, 0x8}}, {0xc000040250, 0x5, ...})
	chezmoi.io/chezmoi/internal/cmd/cmd.go:216 +0x230
chezmoi.io/chezmoi/internal/cmd.Main({{0x1a48764, 0x6}, {0x1a60100, 0x28}, {0x1a4f960, 0x14}, {0x1a49940, 0x8}}, {0xc000040250, 0x5, ...})
	chezmoi.io/chezmoi/internal/cmd/cmd.go:58 +0x277
main.main()
	chezmoi.io/chezmoi/main.go:27 +0xfb
```

running with `--debug` additionally shows this:

```console
time=2025-12-01T16:48:06.898+01:00 level=INFO msg=persistentPreRunRootE version.version=2.67.1 version.commit=5cda5a8a118f2eff0c786c3d45c82669735a346b version.date=2025-11-26T22:53:54Z version.builtBy=Homebrew args="[chezmoi execute-template --override-data {\"foo\":\"baz\"} {{ .foo }} --debug]" goVersion=go1.25.4
time=2025-12-01T16:48:06.898+01:00 level=INFO msg=Stat component=system name=/home/leepa/.config/chezmoi/chezmoistate.boltdb
time=2025-12-01T16:48:06.898+01:00 level=INFO msg=Stat component=system name=/home/leepa/.local/share/chezmoi/.git
time=2025-12-01T16:48:06.901+01:00 level=ERROR msg=ReadFile component=system err="open /home/leepa/.local/share/chezmoi/.chezmoiroot: no such file or directory" name=/home/leepa/.local/share/chezmoi/.chezmoiroot size=0 data=""
time=2025-12-01T16:48:06.902+01:00 level=ERROR msg=ReadFile component=system err="open /home/leepa/.local/share/chezmoi/.chezmoiversion: no such file or directory" name=/home/leepa/.local/share/chezmoi/.chezmoiversion size=0 data=""
time=2025-12-01T16:48:06.902+01:00 level=INFO msg=Close component=persistentState
```

## Output of `chezmoi doctor`

<details>

```console
❯ chezmoi doctor                                                               
RESULT   CHECK                       MESSAGE
ok       version                     v2.67.1, commit 5cda5a8a118f2eff0c786c3d45c82669735a346b, built at 2025-11-26T22:53:54Z, built by Homebrew
ok       latest-version              v2.67.1
ok       os-arch                     linux/amd64 (Fedora Linux 43 (KDE Plasma Desktop Edition))
ok       uname                       Linux tltp 6.17.8-300.fc43.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Nov 14 01:47:12 UTC 2025 x86_64 GNU/Linux
ok       go-version                  go1.25.4 (gc)
ok       executable                  /home/linuxbrew/.linuxbrew/Cellar/chezmoi/2.67.1/bin/chezmoi
ok       upgrade-method              brew-upgrade
ok       config-file                 found ~/.config/chezmoi/chezmoi.toml, last modified 2025-11-05T08:42:15+01:00
ok       source-dir                  ~/.local/share/chezmoi is a git working tree (clean)
ok       suspicious-entries          no suspicious entries
ok       working-tree                ~/.local/share/chezmoi is a git working tree (clean)
ok       dest-dir                    ~ is a directory
error    hardlink                    failed creating hardlink from ~/.local/share/chezmoi to /tmp: link ~/.local/share/chezmoi/.chezmoi-doctor-hardlink-test /tmp/chezmoi-doctor2502605459/.chezmoi-doctor-hardlink-test: invalid cross-device link
ok       symlink                     created symlink from .new-name to .old-name
ok       umask                       022
ok       cd-command                  found /bin/zsh
ok       cd-args                     /bin/zsh
info     diff-command                not set
ok       edit-command                found /usr/bin/vim
ok       edit-args                   /usr/bin/vim
ok       git-command                 found /usr/bin/git, version 2.52.0
ok       merge-command               found /usr/bin/vimdiff
ok       shell-command               found /bin/zsh
ok       shell-args                  /bin/zsh
info     age-command                 age not found in $PATH
ok       gpg-command                 found /usr/bin/gpg, version 2.4.8
info     pinentry-command            not set
info     1password-command           op not found in $PATH
info     bitwarden-command           bw not found in $PATH
info     bitwarden-secrets-command   bws not found in $PATH
info     dashlane-command            dcli not found in $PATH
info     doppler-command             doppler not found in $PATH
info     gopass-command              gopass not found in $PATH
ok       keepassxc-command           found /usr/bin/keepassxc-cli, version 2.7.11
ok       keepassxc-db                ~/path-to-db.kdbx is a file
info     keeper-command              keeper not found in $PATH
info     lastpass-command            lpass not found in $PATH
info     pass-command                pass not found in $PATH
info     passhole-command            ph not found in $PATH
info     rbw-command                 rbw not found in $PATH
info     vault-command               vault not found in $PATH
info     secret-command              not set
```

</details>

## Additional context

This happens on Fedora Linux as in `chezmoi doctor` and on MacOS Tahoe - on both systems chezmoi was installed using homebrew.
I have also downloaded (only for Linux) chezmoi versions 2.66.0-2.67.1 from the releases artifacts - with the same effect.

--- a/internal/cmd/config.go
+++ b/internal/cmd/config.go
@@ -3087,6 +3087,7 @@ func newConfigFile(bds *xdg.BaseDirectorySpecification) ConfigFile {
 		Color: autoBool{
 			auto: true,
 		},
+		Data:         make(map[string]any),
 		Interpreters: defaultInterpreters,
 		Mode:         chezmoi.ModeFile,
 		Pager:        os.Getenv("PAGER"),

--- a/internal/cmd/testdata/scripts/issue4796.txtar
+++ b/internal/cmd/testdata/scripts/issue4796.txtar
@@ -1,0 +1,3 @@
+# test that chezmoi execute-template --override-data does not panic when there is no template data
+exec chezmoi execute-template --override-data '{"foo":"baz"}' '{{ .foo }}'
+stdout baz

